frontend/lib/profile/view/profile.view.dart

```diff
@@ -44,17 +44,47 @@ class _ProfileState extends State<Profile> {
 
         final profile = profileViewModel.profile;
         if (profile == null) {
-          return const Center(child: Text('Error loading profile'));
+          return Center(
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                const Text(
+                  'Error loading profile',
+                  style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
+                ),
+                if (profileViewModel.errorMessage != null)
+                  Padding(
+                    padding: const EdgeInsets.all(16.0),
+                    child: Text(
+                      profileViewModel.errorMessage!,
```

Comment:
Give a generic error page. In general, we never want to expose raw errors in the UI - it's both messy and sometimes dangerous (in practice).